lib/sqlalchemy/dialects/mysql/expression.py

```diff
@@ -0,0 +1,138 @@
+from functools import wraps
+
+from sqlalchemy import exc
+from sqlalchemy.ext.compiler import compiles
+from sqlalchemy.sql.elements import (
+    ColumnElement,
+    ClauseElementBatch,
+)
+
+
+def property_enables_flag(flag_name):
```

Comment:
I thought about changing the object state, but in this case no: we have to create new object to avoid errors. Look the use case example:
```
        match_expr = match(
            users_table.c.firstname,
            users_table.c.lastname,
            against="John Connor",
        )
        stmt = select(users_table)\
            .where(match_expr.in_boolean_mode)\
            .order_by(desc(match_expr))
```
The `WHERE` clause should receive `MATCH` expression with flag `mysql_boolean_mode`, but `ORDER BY` should receive the `MATCH` expression __without__ this flag

And yes: i'm asking to allow to write SQL-like python code